### PR TITLE
Fix patch command osx

### PR DESCRIPTION
**Pull request by Karsten1987**
open for visibility

### CODE REVIEW

File: ros1_rosbag_storage_vendor/CMakeLists.txt [hunk 1]
```diff
@@ -39,20 +39,26 @@ include(ExternalProject)
 # vendor package building against the wrong pluginlib, because the paths get switched.
 # Symptoms of wrong paths: The build of the rosbag for ROS 1 plugins fails with missing symbols
 # like ros::package::getPlugins(...) when linking the vendor package
+if (APPLE)
+  set(PATCH_CMD git apply)
```
Comment (wjwwood) [inline ros1_rosbag_storage_vendor/CMakeLists.txt]: Why is this necessary?

Comment (Karsten1987) [reply ros1_rosbag_storage_vendor/CMakeLists.txt]: When calling only patch on it, on my machine only the following files were patched:

```
patching file tools/rosbag_storage/CMakeLists.txt
patching file tools/rosbag_storage/encryptor_plugins.xml
patching file tools/rosbag_storage/include/rosbag/aes_encryptor.h
patching file tools/rosbag_storage/include/rosbag/encryptor.h
patching file tools/rosbag_storage/include/rosbag/gpgme_utils.h
patching file tools/rosbag_storage/include/rosbag/no_encryptor.h
patching file tools/rosbag_storage/src/aes_encryptor.cpp
patching file tools/rosbag_storage/src/gpgme_utils.cpp
patching file tools/rosbag_storage/src/encryptor.cpp
patching file tools/rosbag_storage/test/test_aes_encryptor.cpp
patching file tools/rosbag_storage/include/rosbag/bag.h
patching file tools/rosbag_storage/src/bag.cpp
patching file tools/rosbag_storage/src/bag.cpp
patching file tools/rosbag_storage/encryptor_plugins.xml
patching file tools/rosbag_storage/package.xml
```
Notably, the file `src/no_encryptor.cpp` did not get created and therefore my build does not compile. I don't really know what exactly the source of this problem is, but with `git apply` everything works fine.

So I can't really tell why the following lines are not being considered:
https://github.com/ros2/rosbag2/blob/master/ros1_rosbag_storage_vendor/resources/rosbag1_encryption_patch.diff#L543-L558

Comment (wjwwood) [reply ros1_rosbag_storage_vendor/CMakeLists.txt]: Hmm, ok, I was just curious. Sometimes I play with the `-p0` vs `-p1` and it helps.
